diff --git a/src/Manifest.php b/src/Manifest.php
index 32c0acd4b5df0db74a1d16b9e7c967cd848c8991..d21778b63a655aab9bbef781aafcb0aa58429211 100644
--- a/src/Manifest.php
+++ b/src/Manifest.php
@@ -71,7 +71,18 @@ class Manifest {
    * Returns styles paths of given chunk.
    */
   public function getStyles(string $chunk, bool $prependBaseUri = TRUE): array {
-    return $this->getChunkPropertyPaths('css', $chunk, $prependBaseUri);
+    if (
+      !$this->chunkExists($chunk)
+      || empty($this->manifest[$chunk]['css'])
+      || !is_array($this->manifest[$chunk]['css'])
+    ) {
+      return [];
+    }
+
+    return array_filter(array_map(
+      fn($import) => $this->getPath($import, $prependBaseUri),
+      $this->manifest[$chunk]['css'],
+    ));
   }
 
   /**
