diff --git a/core/misc/dialog/dialog.position.js b/core/misc/dialog/dialog.position.js
index 986622115..22dd5ad9a 100644
--- a/core/misc/dialog/dialog.position.js
+++ b/core/misc/dialog/dialog.position.js
@@ -63,6 +63,11 @@
    * @fires event:dialogContentResize
    */
   function resetSize(event) {
+    if (!event?.data?.settings || !event?.data?.$element) {
+      return;
+    }
+    const $element = event.data.$element;
+    const settings = event.data.settings;
     const positionOptions = [
       'width',
       'height',
@@ -79,7 +84,7 @@
     let adjustedValue;
     for (let n = 0; n < positionOptions.length; n++) {
       option = positionOptions[n];
-      optionValue = event.data.settings[option];
+      optionValue = settings[option];
       if (optionValue) {
         // jQuery UI does not support percentages on heights, convert to pixels.
         if (
@@ -96,7 +101,7 @@
           // Don't force the dialog to be bigger vertically than needed.
           if (
             option === 'height' &&
-            Math.round(event.data.$element.parent().outerHeight()) <
+            Math.round($element.parent().outerHeight()) <
               adjustedValue
           ) {
             adjustedValue = 'auto';
@@ -106,12 +111,12 @@
       }
     }
     // Offset the dialog center to be at the center of Drupal.displace.offsets.
-    if (!event.data.settings.modal) {
+    if (!settings.modal) {
       adjustedOptions = resetPosition(adjustedOptions);
     }
-    event.data.$element.dialog('option', adjustedOptions);
+    $element.dialog('option', adjustedOptions);
 
-    event.data.$element
+    $element
       ?.get(0)
       ?.dispatchEvent(
         new CustomEvent('dialogContentResize', { bubbles: true }),
