diff --git a/docroot/.htaccess b/docroot/.htaccess
index 4031da475..5d2dba6b9 100755
--- a/docroot/.htaccess
+++ b/docroot/.htaccess
@@ -37,7 +37,11 @@ AddEncoding gzip svgz
 
   # Cache all files for 1 year after access.
   ExpiresDefault "access plus 1 year"
-
+  # Rapid Start document cache length.
+  <FilesMatch "\.(pdf|xls|xlsx|csv|docx|doc|zip)$">
+    ExpiresDefault "access plus 3 hours"
+  </FilesMatch>
+  
   <FilesMatch \.php$>
     # Do not allow PHP scripts to be cached unless they explicitly send cache
     # headers themselves. Otherwise all scripts would have to overwrite the
@@ -59,7 +63,14 @@ AddEncoding gzip svgz
 # Various rewrite rules.
 <IfModule mod_rewrite.c>
   RewriteEngine on
-
+  # Redirect www.water.dcceew.nsw.gov.au -> non-www (must be before www enforcement)
+  RewriteCond %{HTTP_HOST} ^www\.water\.dcceew\.nsw\.gov\.au$ [NC]
+  RewriteRule ^ https://water.dcceew.nsw.gov.au%{REQUEST_URI} [L,R=301]
+  # Enforce HTTPS if not already
+  RewriteCond %{HTTPS} off
+  RewriteCond %{HTTP:X-Forwarded-Proto} !https
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+  
   # Set "protossl" to "s" if we were accessed via https://.  This is used later
   # if you enable "www." stripping or enforcement, in order to ensure that
   # you don't bounce between http and https.
@@ -97,6 +108,12 @@ AddEncoding gzip svgz
   # RewriteCond %{HTTP_HOST} !^www\. [NC]
   # RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
   #
+  RewriteCond %{ENV:AH_SITE_ENVIRONMENT} prod [NC] # only prod
+  RewriteCond %{HTTP_HOST} !\.acquia-sites\.com [NC]
+  RewriteCond %{HTTP_HOST} .
+  RewriteCond %{HTTP_HOST} !^www\. [NC]
+  RewriteCond %{HTTP_HOST} !^water\.dcceew\.nsw\.gov\.au$ [NC]
+  RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
   # To redirect all users to access the site WITHOUT the 'www.' prefix,
   # (http://www.example.com/foo will be redirected to http://example.com/foo)
   # uncomment the following:
