diff --git a/docroot/.htaccess b/docroot/.htaccess
index 4ef72c0..d2e8789 100644
--- a/docroot/.htaccess
+++ b/docroot/.htaccess
@@ -2,6 +2,26 @@
 # Apache/PHP/Drupal settings:
 #
 
+# Redirect hdc URLs to HCCDC domain for all environments.
+RewriteCond %{ENV:AH_SITE_ENVIRONMENT} !prod [NC] # only dev and test
+RewriteCond %{HTTP_HOST} (.+)\.hdc\.nsw\.gov\.au
+RewriteRule (.*) https://%1.hccdc.nsw.gov.au/$1 [R=301,L]
+RewriteCond %{HTTP_HOST} \.acquia-sites\.com [NC]
+RewriteCond %{ENV:AH_SITE_ENVIRONMENT} !prod [NC] # only dev and test
+RewriteCond %{HTTPS} off
+RewriteCond %{HTTP:X-Forwarded-Proto} !https
+RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI} [R=301,L]
+RewriteCond %{ENV:AH_SITE_ENVIRONMENT} prod [NC] # only prod
+RewriteCond %{HTTP_HOST} ^www.hdc.nsw.gov.au [NC,OR]
+RewriteCond %{HTTP_HOST} ^hdc.nsw.gov.au [NC]
+RewriteRule ^(.*)$ https://www.hccdc.nsw.gov.au/$1 [L,R=301]
+RewriteCond %{ENV:AH_SITE_ENVIRONMENT} prod [NC] # only prod
+RewriteCond %{HTTP_HOST} !^www\. [NC]
+RewriteRule ^(.*)$ https://www.hccdc.nsw.gov.au/$1 [L,R=301]
+RewriteCond %{ENV:AH_SITE_ENVIRONMENT} prod [NC] # only prod
+RewriteCond %{HTTPS} off
+RewriteCond %{HTTP:X-Forwarded-Proto} !https
+RewriteRule ^(.*)$ https://www.hccdc.nsw.gov.au$1 [L,R=301]
 # Protect files and directories from prying eyes.
 <FilesMatch "\.(engine|inc|install|make|module|profile|po|sh|.*sql|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock)|web\.config|yarn\.lock|package\.json)$|^#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$">
   <IfModule mod_authz_core.c>
